src/dali/plugin/input/rest/coinbase.py

```diff
@@ -116,53 +126,75 @@ def __init__(
-                if is_fiat(currency):
-                    self.__logger.debug("Skipping fiat transaction: %s", json.dumps(transaction))
-                    continue
-                raw_data: str = json.dumps(transaction)
-                self.__logger.debug("Transaction: %s", raw_data)
-                transaction_type: str = transaction[_TYPE]
-                if transaction_type in {_PRO_DEPOSIT, _PRO_WITHDRAWAL, _EXCHANGE_DEPOSIT, _SEND}:
-                    self._process_transfer(transaction, currency, in_transaction_list, out_transaction_list, intra_transaction_list)
-                elif transaction_type in {_BUY, _SELL, _TRADE}:
-                    self._process_fill(transaction, currency, in_transaction_list, out_transaction_list, id_2_buy, id_2_sell)
-                elif transaction_type in {_INTEREST}:
-                    self._process_interest(transaction, currency, in_transaction_list)
-                elif transaction_type in {_INFLATION_REWARD}:
-                    self._process_income(transaction, currency, in_transaction_list, transaction[_DETAILS][_TITLE])
-                else:
-                    self.__logger.error("Unsupported transaction type (skipping): %s. Please open an issue at %s", raw_data, self.ISSUES_URL)
+        return result
 
-            if in_transaction_list:
-                result.extend(in_transaction_list)
-            if out_transaction_list:
-                result.extend(out_transaction_list)
-            if intra_transaction_list:
-                result.extend(intra_transaction_list)
+    def _process_account(self, account: Dict[str, Any]) -> Optional[TransactionList]:
+        currency: str = account[_CURRENCY][_CODE]
+        account_id: str = account[_ID]
+        in_transaction_list: List[InTransaction] = []
+        out_transaction_list: List[OutTransaction] = []
+        intra_transaction_list: List[IntraTransaction] = []
+
+        self.__logger.debug("Account: %s", json.dumps(account))
+
+        if account[_CREATED_AT] == account[_UPDATED_AT] and RP2Decimal(account[_BALANCE][_AMOUNT]) == ZERO:
+            # skip account without activity to avoid unnecessary API calls
+            return None
+
+        id_2_buy: Dict[str, Any] = {}
+        for buy in self.__get_buys(account_id):
+            id_2_buy[buy[_ID]] = buy
+
+        id_2_sell: Dict[str, Any] = {}
+        for sell in self.__get_sells(account_id):
+            id_2_sell[sell[_ID]] = sell
+
+        for transaction in self.__get_transactions(account_id):
+            if is_fiat(currency):
+                self.__logger.debug("Skipping fiat transaction: %s", json.dumps(transaction))
+                continue
+
+            raw_data: str = json.dumps(transaction)
+            self.__logger.debug("Transaction: %s", raw_data)
+            transaction_type: str = transaction[_TYPE]
+            if transaction_type in {_PRO_DEPOSIT, _PRO_WITHDRAWAL, _EXCHANGE_DEPOSIT, _SEND}:
```

Comment:
_PRIME_WITHDRAWAL is missing: I added it on 3/26. It should generate a merge conflict if you update the source.

Reply:
I don't see _PRIME_WITHDRAWAL being defined in the main branch, are you referring to the following commint? https://github.com/eprbell/dali-rp2/commit/5beff523aa2a4bd5c25ab875de6586b2e6b75f35

Reply:
My mistake: I thought I already pushed it but didn't yet :-) Never mind.